### DISCUSSION TITLE
How to sign Application Bundle for MacOS X on Catalina

### DISCUSSION THREAD

**Discussion by ikus060** [pyinstaller/pyinstaller#6160] [Help]
I tried to follow the recipe to sign code for Mac OS X.  Thanks alot for anyone who wrote it.

1. I'm using my own code signing certificate. The same I'm using for Windows code signing.
2. I've added the certificate to kaychain.
![image](https://user-images.githubusercontent.com/921251/130680377-17b6691f-1fda-43d0-b70e-90d50c7fdcda.png)

I've create a bundle using pyinstaller. That bundle was tested and is working fine.

Then when it code the time to sign the bundle created, is complains about subcomponent.

```
 % codesign -s "Patrik Dufresne Service Logiciel inc." Minarca.app 
Minarca.app: code object is not signed at all
In subcomponent: /Users/vmtest/git/minarca/minarca-client/dist/Minarca.app/Contents/MacOS/lib-dynload/binascii.cpython-39-darwin.so
```

As recommended in the recipe, I've tried with `--deep` but it fail as follow:
```
 % codesign -deep -s "Patrik Dufresne Service Logiciel inc." Minarca.app
codesign: invalid option -- p
Usage: codesign -s identity [-fv*] [-o flags] [-r reqs] [-i ident] path ... # sign
       codesign -v [-v*] [-R=<req string>|-R <req file path>] path|[+]pid ... # verify
       codesign -d [options] path ... # display contents
       codesign -h pid ... # display hosting paths

```
It feel the `-deep` option is missing in my version. Should I upgrade it ?

**Comment by rokm** [comment]
You sure that shouldn't be a `--deep`, with two dashes?

**Reply by ikus060** [reply]
I think you are right. I've been working for too long today. Time to rest
